sources/core/Xenko.Core/Threading/ThreadPool.cs

```diff
@@ -19,109 +20,71 @@ internal class ThreadPool
         public static readonly ThreadPool Instance = new ThreadPool();
 
         private readonly int maxThreadCount = Environment.ProcessorCount + 2;
-        //private readonly int maxThreadCount = Environment.ProcessorCount * 2;
-        private readonly List<Task> workers = new List<Task>();
         private readonly Queue<Action> workItems = new Queue<Action>();
         private readonly ManualResetEvent workAvailable = new ManualResetEvent(false);
 
-        private SpinLock spinLock = new SpinLock();
-        private int activeThreadCount;
+        private object _lock = new object();
+        volatile private int activeThreadCount;
+        volatile private int aliveThreadCount;
 
         public void QueueWorkItem([NotNull] [Pooled] Action workItem)
         {
-            var lockTaken = false;
-            try
+            lock (_lock)
```

Comment:
Locking is very expensive here as @xen2 mentioned. `Spinlock` might be better, contrary to what I thought first.